src/Microsoft.Identity.Web/WebApiAuthenticationBuilderExtensions.cs

```diff
@@ -99,7 +99,9 @@ public static partial class WebApiAuthenticationBuilderExtensions
                 // This is a Microsoft identity platform Web API
                 options.Authority = AuthorityHelpers.EnsureAuthorityIsV2(options.Authority);
 
-                if (options.TokenValidationParameters.AudienceValidator == null)
+                if (options.TokenValidationParameters.AudienceValidator == null
```

Comment:
Only provide a default in the default case - when nothing was set.

Reply:
@jmprieur is your model that you want to exclude any additional audiences?
This might be a good place to pass options.TokenValidationParameters.Clone() as I assume this is called once at setup.